src/olympia/editors/management/commands/auto_approve.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+from collections import Counter
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+from django.core.urlresolvers import reverse
+
+import commonware.log
+
+from olympia import amo
+from olympia.editors.models import (
+    AutoApprovalNotEnoughFilesError, AutoApprovalNoValidationResultError,
+    AutoApprovalSummary)
+from olympia.editors.views import (
+    clear_reviewing_cache, get_reviewing_cache, set_reviewing_cache)
+from olympia.versions.models import Version
+from olympia.zadmin.models import get_config
+
+
+log = commonware.log.getLogger('z.editors.auto_approve')
+
+
+class Command(BaseCommand):
+    help = 'Auto-approve add-ons based on predefined criteria'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            dest='dry_run',
+            default=False,
+            help='Do everything except actually approving add-ons.')
+
+    def fetch_candidates(self):
+        return (Version.objects.filter(
+            addon__type=amo.ADDON_EXTENSION,
+            addon__disabled_by_user=False,
+            addon__status=amo.STATUS_PUBLIC,
+            files__status=amo.STATUS_AWAITING_REVIEW,
+            files__is_webextension=True)
+            .no_cache().order_by('nomination', 'created').distinct())
+
+    def handle(self, *args, **options):
+        dry_run = options.get('dry_run', False)
+        max_average_daily_users = int(
+            get_config('AUTO_APPROVAL_MAX_AVERAGE_DAILY_USERS') or 0)
+        min_approved_updates = int(
+            get_config('AUTO_APPROVAL_MIN_APPROVED_UPDATES') or 0)
+
+        if min_approved_updates <= 0 or max_average_daily_users <= 0:
+            # Auto approval are shut down if one of those values is not present
+            # or <= 0.
+            url = '%s%s' % (
+                settings.SITE_URL,
+                reverse('admin:zadmin_config_changelist'))
+            raise CommandError(
+                'Auto-approvals are deactivated because either '
+                'AUTO_APPROVAL_MAX_AVERAGE_DAILY_USERS or '
+                'AUTO_APPROVAL_MIN_APPROVED_UPDATES have not been '
+                'set or were set to 0. Use the admin tools Config model to '
+                'set them by going to %s.' % url)
+
+        stats = Counter()
+        qs = self.fetch_candidates()
+        stats['total'] = len(qs)
+
+        successful_verdict = (
+            amo.WOULD_HAVE_BEEN_AUTO_APPROVED if dry_run
+            else amo.AUTO_APPROVED)
+
+        for version in qs:
+            # Is the addon already locked by a reviewer ?
+            if get_reviewing_cache(version.addon.pk):
+                stats['locked'] += 1
+                continue
+
+            # Lock the addon for ourselves, no reviewer should touch it.
+            set_reviewing_cache(version.addon.pk, settings.TASK_USER_ID)
```

Comment:
I was a bit paranoid of how long it could take. Also, ultimately this might be run several times per hour, so the chance of a conflict increases.